### PR TITLE
BXMSPROD-1231 return varibles by pmebuild script

### PR DESCRIPTION
**Pull request by Ginxo**
https://issues.redhat.com/browse/BXMSPROD-1231

- https://github.com/kiegroup/kogito-pipelines/pull/344
- https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/180

### CODE REVIEW

File: vars/pmebuild.groovy [hunk 1]
```diff
@@ -45,14 +51,15 @@ def buildProject(String project, String settingsXmlId, Map<String, Object> build
         executePME(finalProjectName, projectConfig, pmeCliPath, settingsXmlId, variableVersionsMap)
         executeBuildScript(finalProjectName, buildConfig, settingsXmlId, "-DaltDeploymentRepository=local::default::file://${env.WORKSPACE}/deployDirectory")
 
+        def key = projectVariableMap[group + '_' + name]
```
Comment (radtriste) [inline vars/pmebuild.groovy:54]: what happens if projectVariableMap does not contain the `group + '_' + name` ?

File: vars/pmebuild.groovy [hunk 2]
```diff
@@ -19,9 +20,13 @@ def buildProjects(List<String> projectCollection, String settingsXmlId, String b
     Map<String, Object> buildConfigMap = getBuildConfiguration(buildConfigContent, buildConfigPathFolder, buildConfigAdditionalVariables)
   
     checkoutProjects(projectCollection, buildConfigMap, buildConfigAdditionalVariables)
-    projectCollection.each { project -> buildProject(project, settingsXmlId, buildConfigMap, pmeCliPath, projectVariableMap, variableVersionsMap) }
+    def result = projectCollection.inject([:]) { acc, project ->
+        acc[project] = buildProject(project, settingsXmlId, buildConfigMap, pmeCliPath, projectVariableMap, variableVersionsMap)
+        acc
+    }
```
Comment (radtriste) [inline vars/pmebuild.groovy:26]: Could we use `collectEntries` here ?

```suggestion
    def result = projectCollection.collectEntries { project ->
        return [ (project) : buildProject(project, settingsXmlId, buildConfigMap, pmeCliPath, projectVariableMap, variableVersionsMap) ]
    }
```

File: vars/pmebuild.groovy (hunk 1 above)
Comment (Ginxo) [reply vars/pmebuild.groovy:54]:
```suggestion
```

File: vars/pmebuild.groovy [hunk 3]
```diff
@@ -45,14 +51,15 @@ def buildProject(String project, String settingsXmlId, Map<String, Object> build
         executePME(finalProjectName, projectConfig, pmeCliPath, settingsXmlId, variableVersionsMap)
         executeBuildScript(finalProjectName, buildConfig, settingsXmlId, "-DaltDeploymentRepository=local::default::file://${env.WORKSPACE}/deployDirectory")
 
+        def key = projectVariableMap[group + '_' + name]
+        def pom = readMavenPom file: 'pom.xml'
+        result = pom?.version
         if (projectVariableMap.containsKey(group + '_' + name)) {
```
Comment (Ginxo) [inline vars/pmebuild.groovy:57]:
```suggestion
        if (projectVariableMap.containsKey("${group}_${name}")) {
```

File: vars/pmebuild.groovy [hunk 4]
```diff
@@ -45,14 +51,15 @@ def buildProject(String project, String settingsXmlId, Map<String, Object> build
...
-            def key = projectVariableMap[group + '_' + name]
-            def pom = readMavenPom file: 'pom.xml'
-            variableVersionsMap << ["${key}": pom.version]
+            variableVersionsMap << ["${key}": result]
```
Comment (Ginxo) [inline vars/pmebuild.groovy:58]:
```suggestion
            def key = projectVariableMap["${group}_${name}"]
            variableVersionsMap << ["${key}": result]
```

**Review by radtriste — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Pull request merged**